components/markdown/markdown.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import messageHtmlToComponent from 'utils/message_html_to_component';
+import EmojiMap from 'utils/emoji_map';
+import * as TextFormatting from 'utils/text_formatting';
+
+type Options = {
+    mentionHighlight?: boolean;
+}
+
+type Props = {
+
+    /*
+     * An object mapping channel names to channels for the current team
+     */
+    channelNamesMap?: object;
+
+    /*
+     * An array of URL schemes that should be turned into links. Anything that looks
+     * like a link will be turned into a link if this is not provided.
+     */
+    autolinkedUrlSchemes?: Array<any>;
+
+    /*
+     * Whether or not to do Markdown rendering
+     */
+    enableFormatting?: boolean;
+
+    /*
+     * Whether or not this text is part of the RHS
+     */
+    isRHS?: boolean;
+
+    /*
+     * An array of words that can be used to mention a user
+     */
+    mentionKeys?: Array<object>;
+
+    /*
+     * The text to be rendered
+     */
+    message: string;
+
+    /*
+     * Any additional text formatting options to be used
+     */
+    options: Options;
+
+    /*
+     * The root Site URL for the page
+     */
+    siteURL?: string;
+
+    /*
+     * The current team
+     */
+    team?: object;
+
+    /**
+     * If an image proxy is enabled.
+     */
+    hasImageProxy?: boolean;
+
+    /**
+     * Minimum number of characters in a hashtag.
+     */
+    minimumHashtagLength?: number;
+
+    /**
+     * Whether or not to proxy image URLs
+     */
+    proxyImages?: boolean;
+
+    /**
+     * Any extra props that should be passed into the image component
+     */
+    imageProps?: object;
+
+    /**
+     * prop for passed down to image component for dimensions
+     */
+    imagesMetadata?: object;
+
+    /**
+     * Whether or not to place the LinkTooltip component inside links
+     */
+    hasPluginTooltips?: boolean;
+
+    /**
+     * Post id prop passed down to markdown image
+     */
+    postId?: string;
+
+    /**
+     * Post id prop passed down to markdown image
+     */
+    postType?: string;
```

Comment:
And this can is of the type `PostType` from `mattermost-redux/types/posts` as well.